elastic/apm-server
# Merge ES default values to Sourcmeap configuration.

## Motivation/summary
When configuring an Elasticsearch instance for Sourcemap usage, Elasticsearch default values should be merged with the custom config, to avoid having to configure all settings.

At the moment default values are not merged as soon as any of the `apm-server.rum.source_mapping.elasticsearch.*` options is configured, which is unexpected behavior and not aligned with how other configuration options are merged with default values. 

This PR changes the behavior to use default options where no dedicated configuration option is provided. 

## Checklist

~- [ ] I have signed the [Contributor License Agreement](https://www.elastic.co/contributor-agreement/).~

~- [ ] I have made corresponding changes to the documentation~


## How to test these changes

Configure `apm-server.rum.source_mapping.elasticsearch.*` without defining a `host`. It should be merged with the default host. 
Default values are `hosts: []string{"localhost:9200"}, protocol: "http", timeout: 5s`. 

Also configure with a different host than the default host and ensure it is applied. 

Without these changes, the default values would not be set and an error would occur. 

## Related issues

fixes #3350
